app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivityPresenter.kt

```diff
@@ -19,188 +19,188 @@
-     *
-     * @param stackIsClear Boolean state of fragments stack
-     */
-    fun onFragmentBackStackChanged(stackIsClear: Boolean, intent: Intent?) {
-        viewContract.updateNavigationViewsContent()
-        intent?.let {
-            if (stackIsClear && isInstantActionIntent(it)) viewContract.closeView()
-        }
-    }
-
-    /**
-     * Handles clicks on navigation action (back arrow)
-     *
-     * @param stackIsClear Boolean state of fragments stack
-     */
-    fun onNavigationItemClick(stackIsClear: Boolean) {
-        if (stackIsClear) viewContract.closeView() else viewContract.popBackStack()
-    }
-
-    /**
-     * Connection was selected in Connection selector
-     */
-    fun onConnectionSelected(connectionGuid: String) {
-        try {
-            viewContract.showSubmitActionFragment(connectionGuid, actionAppLinkData!!)
-        } catch (e:Exception) {
-            e.log()
-        }
-    }
-
-    private val Intent?.connectionId: String
-        get() = this?.getStringExtra(KEY_CONNECTION_ID) ?: ""
-
-    private val Intent?.authorizationId: String
-        get() = this?.getStringExtra(KEY_AUTHORIZATION_ID) ?: ""
-
-    private val Intent?.deepLink: String
-        get() = this?.getStringExtra(KEY_DEEP_LINK) ?: ""
-
-    // Show Authorization Details Fragment
-    private val Intent?.hasPendingAuthorizationData: Boolean
-        get() = this != null && this.connectionId.isNotEmpty() && this.authorizationId.isNotEmpty()
-
-    // Show Connect Activity
-    private val Intent?.hasDeepLinkData: Boolean
-        get() = deepLink.isNotEmpty()
-
-    private fun onConnectAppLinkDataReceived(connectionAppLinkData: ConnectAppLinkData) {
-        viewContract.setSelectedTabbarItemId(R.id.menu_connections)
-        viewContract.showConnectProvider(connectionAppLinkData)
-    }
-
-    private fun onActionAppLinkDataReceived(actionAppLinkData: ActionAppLinkData) {
-        this.actionAppLinkData = actionAppLinkData
-        val connections = connectionsRepository.getByConnectUrl(actionAppLinkData.connectUrl)
-        when {
-            connections.isEmpty() -> {
-                viewContract.showNoConnectionsError()
-            }
-            connections.size == 1 -> {
-                val connectionGuid = connections.first().guid
-                viewContract.showSubmitActionFragment(
-                    connectionGuid = connectionGuid,
-                    actionAppLinkData = actionAppLinkData
-                )
-            }
-            else -> {
-                val result = connections.convertConnectionsToViewModels(
-                    context = appContext
-                )
-                viewContract.showConnectionsSelectorFragment(result)
-            }
-        }
-    }
-
-    private fun isInstantActionIntent(intent: Intent): Boolean {
-        return intent.hasPendingAuthorizationData || intent.deepLink.extractActionAppLinkData() != null
-    }
-}
+//TODO REMOVE AFTER IMPLEMENTING OF ALL CHILD COMPONENTS
```

Comment:
Can you list exactly what elements are left?

Reply:
What you mean?

Reply:
discussed